app/main.py

```diff
@@ -1,34 +1,57 @@
+from typing import Dict, Tuple, List
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    battle_array: List[List[str]] = []
+
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
+        self.battle_array = [["~" for _ in range(self.row)]
+                             for _ in range(self.column)]
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: Tuple[int, int],
+                 end: Tuple[int, int],
+                 is_drowned: bool = False
+                 ) -> None:
+        self.decks = []
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        for row in range(start[0], end[0] + 1):
+            for column in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row, column))
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, row: int, column: int) -> None:
+        self.get_deck(row, column).is_alive = False
+        self.is_drowned = all(not deck.is_alive for deck in self.decks)
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self,
+                 ships: List[Tuple[Tuple[int, int], Tuple[int, int]]]
```

Comment:
or you can use built-in types
```suggestion
                 ships: list[tuple[tuple[int, int], tuple[int, int]]]
```